### PR TITLE
Add API for pkcs12_derivation

### PR DESCRIPTION
**Pull request by jack-fortanix**

### CODE REVIEW

File: mbedtls/src/hash/mod.rs [hunk 1]
```diff
@@ -167,18 +166,42 @@ pub fn pbkdf2_hmac(
...
+            iterations,
+            key.len() as u32,
+            key.as_mut_ptr()
+        )
+        .into_result());
+        Ok(())
+    }
+}
+
+pub fn pkcs12_pbkdf(
```
[the remark this answers is not in the document]
Comment (jack-fortanix) [reply mbedtls/src/hash/mod.rs]: Renamed.

**Comment by jack-fortanix** [comment]
What's bors waiting for here?

**Comment by jethrogb** [comment]
@jack-fortanix for someone to talk to it :)

bors r+

**Comment by jack-fortanix** [comment]
Ah ok I thought bors would see the approved status on github.

**Pull request merged**